src/controllers/components/student-panel.js

```diff
@@ -8,4 +8,21 @@ export default class StudentPanelController{
             Dispatcher.handleAction("MODELS_RECEIVED", models);
         });
     }
+
+    static countDuplicateModels(models) {
+        let modelCounts = [];
+        models.forEach(model => {
+            let exists = false;
+            modelCounts.forEach(mc => {
+                if(model.address === mc.address && !exists){
+                    mc.quantity++;
+                    exists = true;
+                }
+            });
+            if(!exists){
+                modelCounts.push({address: model.address, name: model.name, quantity: 1});
+            }
+        });
+        return modelCounts;
```

Comment:
Also, this could be slightly simplified to something like...

```javascript
return models.reduce((counts, model) => {
    let index = counts.findIndex(count => count.address === model.address);
    if (index < -1) {
        counts.push({address: model.address, name: model.name, quantity: 1});
    } else {
        counts[index].quantity ++;
    }
    return counts;
}, []);
```